src/main/perl/Fetch.pm

```diff
@@ -181,43 +173,48 @@ sub fetchProfile
 
     my $profile = $self->download("profile");
 
+    my $res;
     if (!defined($profile)) {
         $self->error("Failed to fetch profile $self->{PROFILE_URL}");
-        return undef;
-    }
-
-    return SUCCESS unless $profile;
-
-    local $SIG{__DIE__} = sub {
-        warn "Cleaning on die";
+    } elsif ("$profile" eq "0") {
+        # Have to do string comparison, because profile can be a FileReader or 0
+        $self->verbose("fetchProfile: nothing to do, ",
+                       "profile is not more recent than local cache (and no FORCE)");
+        $res = SUCCESS;
+    } else {
+        local $SIG{__DIE__} = sub {
+            warn "Cleaning on die";
+            $self->_cleanup(\%current, \%previous, 1);
+            confess(@_);
+        };
+
+        $self->verbose("Downloaded new profile");
+
+        %current = $self->current($profile, %previous);
+        if ($self->process_profile("$profile", %current) == $ERROR) {
+            $self->error("Failed to process profile for $self->{PROFILE_URL}");
+            $self->_cleanup(\%current, \%previous, 1);
+            return $ERROR;
+        }
+
+        # Make the new profile/CID the latest.cid
+        my $new_cid = "$current{cid}";
+
+        $previous{cid}->set_contents($new_cid);
```

Comment:
the comment above the previous methods says `# Previous is a bit of a misnomer. This is about the "latest.cid"`. this code updates latest.cid . i only cleaned this code because i was trying to find out why simple changes broke unittests at seemingly random places. i'm only the victim here :smile: